### ISSUE TITLE
[BUG] ContOS 7 / CentOS8 rspamd_symcache_load_items: cannot open file /var/lib/rspamd/symbols.cache

### ISSUE THREAD

**Issue by whataboutpereira** [rspamd/rspamd#3086]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**

Rspamd cannot open file /var/lib/rspamd/symbols.cache

### Steps to Reproduce

1. Install a clean CentOS 8 system
2. Install Rspamd 2.0 from the official Rspamd repo
3. Start rspamd
4. Observe the error message in rspamd.log

2019-10-15 11:10:54 #561(main) <q1wgw1>; symcache; rspamd_symcache_load_items: cannot open file /var/lib/rspamd/symbols.cache, error 2, No such file or directory
2019-10-15 11:14:48 #564(normal) <48cdd1>; lua; common.lua:223: got error checking cache: timeout while connecting the server

For testing I created the empty file and chmod _rspamd:_rspamd, which resulted in:

2019-10-15 11:57:16 #648(main) <q1wgw1>; symcache; rspamd_symcache_load_items: cannot use file /var/lib/rspamd/symbols.cache, error 22, Invalid argument

**Expected behavior**

No error.

### Versions

Rspamd daemon version 2.0

OS: CentOS 8 running in the latest Proxmox (cpu: Sandy Bridge)
Linux 4.18.0-80.el8.x86_64

Same problem exists when running Rspamd 2.0 in a CentOS 7 Linux Container:

OS: CentOS 7 running in the latest Proxmox (cpu: Sandy Bridge)
Kernel 5.0.21-2-pve on an x86_64

**Comment by vstakhov** [comment]
There should be something wrong with the permissions on `/var/lib/rspamd`. I don't really think it's Rspamd issue. You should definitely not create empty files for that: if Rspamd cannot find cache file, it will just continue.

**Comment by whataboutpereira** [comment]
stat /var/lib/rspamd
  File: /var/lib/rspamd
  Size: 12288           Blocks: 24         IO Block: 4096   directory
Device: fd00h/64768d    Inode: 263998      Links: 2
Access: (0755/drwxr-xr-x)  Uid: (  994/ _rspamd)   Gid: (  991/ _rspamd)
Context: system_u:object_r:var_lib_t:s0
Access: 2019-10-15 18:36:19.072698240 +0300
Modify: 2019-10-15 18:36:30.301719104 +0300
Change: 2019-10-15 18:36:30.301719104 +0300

I just rested with SELinux enforcing off, but no change. I'm glad it's not critical.

I'm getting all sorts of failed DKIM and ARC checks from incoming mail from Google, etc. with 2.0 and this is the only "error" I was seeing.

**Comment by vstakhov** [comment]
What is your log level?

**Comment by whataboutpereira** [comment]
I'm seeing it with the default (info).

**Comment by hboetes** [comment]
> There should be something wrong with the permissions on `/var/lib/rspamd`. I don't really think it's Rspamd issue. You should definitely not create empty files for that: if Rspamd cannot find cache file, it will just continue.

I'm getting the same error on OpenBSD 6.8 whilst running rspamd 2.6.

```
<nk7j81>; symcache; rspamd_symcache_load_items: cannot open file /var/rspamd/symbols.cache, error 2, No such file or directory
```

And it's really not a permission problem.
```
% ls -ld /var/rspamd/    
drwxr-xr-x 2 _rspamd _rspamd 1.0K Dec 22 00:32 /var/rspamd/
% ls -ld /var/rspamd/symbols.cache
gls: cannot access '/var/rspamd/symbols.cache': No such file or directory
```

> cache_file: used to store information about rules and their statistics; this file is automatically generated if Rspamd detects that a symbol’s list has been changed.

Should it really be logged as an error if this file does not exist?

**Comment by ChR-iSz** [comment]
Same here... Any news ?

`cannot create file /var/lib/rspamd/symbols.cache.new:`

**Comment by innogrey** [comment]
Anything new in this matter? FreeBSD 13.1, rspamd-3.4_1 and I just encountered this:
rspamd[38802]: <6zqsuu>; symcache; save_items: cannot create file /var/db/rspamd/symbols.cache.new: No such file or directory

**Comment by Artim96** [comment]
Same phenomenon here on Debian 10 with rspamd 3.4. You can manually create /var/lib/rspamd/symbols.cache.new but it won't survive a restart of the service. And I don't think it's the best idea to make the file undeletable.

**Comment by schmidp** [comment]
I am also seeing this on Ubuntu 22.04.

**Comment by allanwind** [comment]
Wasn't this fixed in #4337?
